src/Samples/NewWindow/Program.fs

```diff
@@ -112,12 +112,21 @@ module App =
   ]
 
 
-[<EntryPoint; STAThread>]
-let main _ =
-  let createWindow2 () = Window2(Owner = Application.Current.MainWindow)
-  let bindings = App.mainBindings Window1 createWindow2
+let fail _ = failwith "never called"
+let mainVm = ViewModel.designInstance (App.init ()) (App.mainBindings fail fail ())
+let window1Vm = ViewModel.designInstance (App.init ()) (App.window1Bindings ())
+let window2Vm = ViewModel.designInstance App.initWindow2 (App.window2Bindings ())
+
+
+let main (mainWindow, (createWindow1: Func<#Window>), (createWindow2: Func<#Window>)) =
+  let createWindow1 () = createWindow1.Invoke()
+  let createWindow2 () =
+    let window = createWindow2.Invoke()
+    window.Owner <- mainWindow
+    window
```

Comment:
An alternative is to use type parameters here:

```suggestion
let main<'Win1, 'Win2 when 'Win1: (new: unit -> 'Win1) and 'Win2: (new: unit -> 'Win2) and 'Win1 :> Window and 'Win2 :> Window> mainWindow =
  let createWindow1 () = new 'Win1()
  let createWindow2 () =
    let window = new 'Win2()
    window.Owner <- mainWindow
    window
```

(Modification to C# code also needed, of course.)

The difference is that the type parameter approach guarantees that this code controls the construction, as opposed to the C# code being able to pass a function that does other stuff, too. It's slightly more verbose, though.

Use the approach you prefer.

Reply:
This is interesting.  I forgot about this approach.  I do like that it avoids having to deal with `Func<>` in F#, but the additional verbosity without any need for the extra safety (since it is trivial to check that it is essentially constructors being passed, and it is also acceptable if the C# code sent a more complex function anyway) makes me prefer passing in `Func<>`s.  I will resolve without change then since you deferred to me, but if you change your mind though, then I am also fine with the type parameter solution.

Reply:
I'm fine with the current approach.